contracts/DaiHrd.sol

```diff
@@ -464,4 +504,99 @@ contract DaiHrd is ERC777 {
 		vat.move(address(this), msg.sender, attorontodai);
 		return attorontodai;
 	}
+
+	// Dai specific functions. These functions all behave similar to standard functions with input or output denominated in Dai instead of DaiHrd
+	function balanceOfDai(address tokenHolder) external view returns(uint256 attodai) {
+		uint256 rontodaiPerPot = calculatedChi();
+		uint256 attodaiHrd = balanceOf(tokenHolder);
+		return convertAttodaiHrdToAttodai(attodaiHrd, rontodaiPerPot);
+	}
+
+	function totalSupplyDai() external view returns(uint256 attodai) {
+		uint256 rontodaiPerPot = calculatedChi();
+		attodai = convertAttodaiHrdToAttodai(totalSupply(), rontodaiPerPot);
+		return attodai;
+	}
+
+	function withdrawDai(uint256 attodai) external returns(uint256 attodaiHrd) {
```

Comment:
Did we come to agreement on this function taking an `address`?  Argument for that change was that there is really no reason to withdraw DAI unless you are going to immediately be sending it away.

Reply:
tell you what, if we add it, i want to check against 0x0, address(this), the dai address

Reply:
The naming on this one is a bit weird since both withdraw functions withdraw DAI.  I wonder if we should rename `withdraw` to `withdrawDaiDenominatedInDaiHrd` and this one would just be `withdrawDai`?  Ugh.  Naming is hard.

Reply:
Or, if we wanted consistency:
```
withdrawDai(attodaiHrd)
withdrawDaiDenominatedInDai(attodai)
```

Reply:
I support `require(destination != address(0))` 💯